source/frontend/src/components/maps/leaflet/Control/LayersControl/LayersMenu.tsx

```diff
@@ -83,21 +83,54 @@ const LayerColor = styled.div<{ color: string }>`
 const ParentCheckbox: React.FC<
   React.PropsWithChildren<{ name: string; label: string; index: number }>
 > = ({ name, label, index }) => {
-  const { values, setFieldValue } = useFormikContext();
+  const { values, setFieldValue } = useFormikContext<LayersFormModel>();
 
   const onChange = () => {
     const nextValue = !getIn(values, name);
-    setFieldValue(name, nextValue);
-    // Toggle children nodes
+    // Toggle children nodes if parent is selected/deselected
     const nodes = getIn(values, `layers[${index}].nodes`) || [];
-    nodes.forEach((node: any, i: number) =>
-      setFieldValue(`layers[${index}].nodes[${i}].on`, nextValue),
+    nodes.forEach((_: TreeNodeInArray, childIndex: number) =>
+      setFieldValue(`layers[${index}].nodes[${childIndex}].on`, nextValue),
     );
   };
 
+  // Calculate the state of the parent check based on all the children.
+  useEffect(() => {
+    const parentLayer = values.layers[index];
+    const activeChildren = parentLayer?.nodes?.filter(x => x?.on === true);
+    let isActive = false;
+    let isPartialActive = false;
+
+    if (activeChildren?.length > 0) {
+      isActive = true;
+      if (parentLayer?.nodes?.length !== activeChildren?.length) {
+        isPartialActive = true;
+      } else {
```

Comment:
this seems redundant given that isPartialActive is set every time this useEffect runs.